From 902836ece2a6524ef6ec3759b9e80765c7cf60b9 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 22 Aug 2022 13:46:55 +0300
Subject: [PATCH] [SW-86869] gaudi3: modify valid fw version check

Currently, ROM only check whether firmware version ie. svn received
as part of image manifest is higher or equal to minimum svn read
from efuse, however there is possibility firmware version received
as part of firmware can be above 128 which will be allowed to boot
by ROM. This patch modifies check to verify svn above 128 as well. This
is recommendation from security team to verify svn above 128.

Change-Id: Iefe7863e6039c2c0a99fcb1a81f43e207f2c7439
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index a258183..bfe6219 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -683,10 +683,12 @@
 
 int security_check_valid_fw_version(uint32_t fw_version)
 {
-	if (fw_version >= arcboot_rom_global.sec_data.sec_ver)
-		return SECURITY_FW_VERSION_SECURE;
+	if ((fw_version < arcboot_rom_global.sec_data.sec_ver) ||
+		(fw_version > SECURITY_MAX_FW_SEC_VERSION)) {
+		return SECURITY_FW_VERSION_NOT_SECURE;
+	}
 
-	return SECURITY_FW_VERSION_NOT_SECURE;
+	return SECURITY_FW_VERSION_SECURE;
 }
 
 int security_efuse_tpm_en(void)
